src/Engine/Animation/AnimationController.cpp

```diff
@@ -71,14 +88,37 @@ void AnimationController::Load(const std::string& name) {
         }
     }
 
+    uint32_t numBools = boolMap.size();
+    file.read(reinterpret_cast<char*>(&numBools), sizeof(uint32_t));
+    boolMap.resize(numBools);
+    for (auto i = 0; i < numBools; ++i) {
+        file.read(reinterpret_cast<char*>(boolMap[i]), sizeof(BoolItem));
+    }
+
+    uint32_t numFloats = boolMap.size();
+    file.read(reinterpret_cast<char*>(&numFloats), sizeof(uint32_t));
+    boolMap.resize(numFloats);
+    for (auto i = 0; i < numFloats; ++i) {
+        file.read(reinterpret_cast<char*>(boolMap[i]), sizeof(FloatItem));
+    }
+
     // Close file.
     file.close();
 }
 
-void Animation::AnimationController::Clear() {
+void AnimationController::Clear() {
     for (Node* node : animationNodes) {
         delete node;
     }
-
     animationNodes.clear();
+
+    for (BoolItem * b : boolMap) {
```

Comment:
Asterisk goes to the left. Braces around single statement. Also applies to floats below.